### PR TITLE
Issue 32 - Use success request for swagger examples

### PR DESCRIPTION
**Pull request by Danielwsx64**
When merging `Path Item Objects` use the success request to generate parameters and req body examples.

Ref: #32

### CODE REVIEW

File: test/support/samples/swagger_formater.ex [hunk 1]
```diff
@@ -1,176 +1,248 @@
...
-            example: "590eda29-df5f-4244-8159-7bb87e3cc15a"
-          }
-        ],
-        responses: %{
-          404 => %{
-            description: "",
-            content: %{
-              "application/json" => %{
-                schema: %{properties: %{"message" => %{type: "string"}}, type: "object"}
+  defmodule PathItemObject do
```
Comment (thiamsantos) [inline test/support/samples/swagger_formater.ex:2]: :nail_care: Seems a little bit strange to have 2 defmodule in the same file, when they fill this many lines.

File: lib/swagger/formatter.ex [hunk 2]
```diff
@@ -134,46 +134,68 @@ defmodule Xcribe.Swagger.Formatter do
...
+    |> Map.update(:parameters, params, &merge_parameter_object_lists(&1, params, mode))
     |> Map.update(:responses, resp, &Map.merge(&1, resp))
-    |> merge_request_body_if_needed(item_two)
+    |> merge_request_body_if_needed(new_item, mode)
   end
 
-  defp merge_request_body_if_needed(%{requestBody: _} = item, %{requestBody: new_body}) do
-    Map.update(item, :requestBody, new_body, &merge_request_body(&1, new_body))
+  defp overwrite_mode(responses) do
+    code = responses |> Map.keys() |> List.first()
```
Comment (brosquinha) [inline lib/swagger/formatter.ex:165]: Is `code` always the first Map key of Response? Is there any other way to make it clearer that we're aiming at that particular key? :thinking:

Comment (Danielwsx64) [reply lib/swagger/formatter.ex:165]: The struct is like that:

```elixir
      expected = %{
        "get" => %{
          description: "",
          parameters: [],
          security: [%{"api_key" => []}],
          summary: "",
          tags: ["users"],
          responses: %{
            200 => %{description: "", headers: , content: %{  "application/json" => %{ schema: } } }
          }
        }
      }
```
The response map have the requests status code as keys. As we generate one path at a time the map has just one key. I don't know what key is coming, but I know it is just one status code.

File: test/support/samples/swagger_formater.ex (hunk 1 above)
Comment (Danielwsx64) [reply test/support/samples/swagger_formater.ex]: Removed !!

File: test/support/samples/swagger_formater.ex [hunk 3]
```diff
@@ -1,5 +1,5 @@
-defmodule Xcribe.Support.Samples.SwaggerFormater do
-  def path_item_object_without_request_body do
+defmodule Xcribe.Support.Samples.SwaggerFormater.PathItemObject do
```
Comment (thiamsantos) [inline test/support/samples/swagger_formater.ex:1]: Could we also rename the file?

`mv test/support/samples/swagger_formater.ex test/support/samples/swagger_formater/path_item_object.ex`

Comment (Danielwsx64) [reply test/support/samples/swagger_formater.ex:1]: Done :P

**Pull request merged**